### ISSUE TITLE
All pages return 404. CSRF token not found

### ISSUE THREAD

**Issue by jellykells** [pixelfed/support#63]
After following the installation instructions at https://docs.pixelfed.org/ Pixelfed seems to be running fine, but I can't actually access any pages. Resources all return 200, but the document itself returns 404. The browser console shows the error: `CSRF token not found.`

I'm using nginx and the recommended versions of all software. I'm fairly certain all dependencies have been met. My nginx configuration matches the one provided with the exception of the root and PHP socket path.

I'm also handling the 301 redirect to secure connection with a separate nginx reverse proxy, which lies in front of all my web services and seems to be functioning correctly.

**Comment by jellykells** [comment]
I've upgraded from PHP 7.3 to PHP 7.4 and I now receive some (hopefully helpful) output when PHP is started:
>Dec 19 04:09:02 Ubuntu-VMPIX sessionclean[4917]: PHP Warning:  PHP Startup: Unable to load dynamic library 'pdo_mysql' (tried: /usr/lib/php/20190902/pdo_mysql (/usr/lib/php/20190902/pdo_mysql: cannot open shared object file: No such file or directory), /usr/lib/php/20190902/pdo_mysql.so (/usr/lib/php/20190902/pdo_mysql.so: undefined symbol: pdo_parse_params)) in Unknown on line 0

The file does exist:
`-rw-r--r-- 1 root root  31K Nov 28 01:27 pdo_mysql.so`

**Comment by jellykells** [comment]
I copied the additional headers found here https://github.com/osTicket/osTicket/issues/736#issuecomment-71351921 which solves the CSRF error (but only on the main page?), and the main page returns 200.

However, I'm still getting the PHP error I described above, even after removing and reinstalling PHP and MySQL completely. Obviously Pixelfed can't do anything meaningful without a working db connection.

Listing versions here:
Ubuntu 18.04.3
PHP 7.4.0 from ppa:ondrej/php
MySQL 5.7.28
Nginx 1.14.0
